### PR TITLE
General: Show user who created the workfile on the details pane of workfile manager

### PR DESCRIPTION
**Pull request by fabiaserra**
## Changelog Description
New PR for https://github.com/ynput/OpenPype/pull/5087, which was closed after merging `next-minor` branch and then realizing we don't need to target it as it was decided it's not required to support windows. More info on that PR discussion.

Small addition to add name of the `user` who created the workfile on the details pane of the workfile manager:
Screenshot 2023-05-17 161521

## Testing notes:
1. Open workfile manager
2. You should see on the details pane a `User` entry showing who created that file

### CODE REVIEW

File: openpype/tools/workfiles/window.py [hunk 1]
```diff
@@ -134,7 +147,9 @@ def set_context(self, asset_id, task_name, filepath, workfile_doc):
             "<b>Created:</b>",
             creation_time.strftime(datetime_format),
             "<b>Modified:</b>",
-            modification_time.strftime(datetime_format)
+            modification_time.strftime(datetime_format),
```
Comment (fabiaserra) [inline openpype/tools/workfiles/window.py:150]: realizing we don't even want to show user if not Windows, changing it now

**Comment by mkolar** [comment]
Can't test outside of windows at the moment, but the code is looking good and it behaves correctly in windows. Meaning nothing changes.

**Comment by fabiaserra** [comment]
Awesome, thank you @mkolar for testing!

File: openpype/tools/workfiles/window.py [hunk 2]
```diff
@@ -94,6 +95,18 @@ def _on_save_click(self):
...
         self.save_clicked.emit()
 
+    def get_user_name(self, file):
+        """Get user name from file path"""
+        # Only run on Unix because pwd module is not available on Windows.
+        # NOTE: we tried adding "win32security" module but it was not working
+        # on all hosts so we decided to just support Linux until migration
+        # to Ayon
+        if platform.system() != "Windows":
+            import pwd
```
Comment (iLLiCiTiT) [inline openpype/tools/workfiles/window.py:105]: Will this work on MacOS?

File: openpype/tools/workfiles/window.py [hunk 3]
```diff
@@ -94,6 +95,18 @@ def _on_save_click(self):
...
+
+            filestat = os.stat(file)
+            return pwd.getpwuid(filestat.st_uid).pw_name
```
Comment (iLLiCiTiT) [inline openpype/tools/workfiles/window.py:108]:
```suggestion
        if platform.system() != "linux":
            return None
        import pwd

        filestat = os.stat(file)
        return pwd.getpwuid(filestat.st_uid).pw_name
```

File: openpype/tools/workfiles/window.py [hunk 4]
```diff
@@ -134,8 +147,13 @@ def set_context(self, asset_id, task_name, filepath, workfile_doc):
...
             creation_time.strftime(datetime_format),
             "<b>Modified:</b>",
-            modification_time.strftime(datetime_format)
+            modification_time.strftime(datetime_format),
         )
+        if platform.system() != "Windows":
+            lines += (
+                "<b>User:</b>",
+                self.get_user_name(filepath),
+            )
```
Comment (iLLiCiTiT) [inline openpype/tools/workfiles/window.py:156]:
```suggestion
        username = self.get_user_name(filepath)
        if username:
            lines += (
                "<b>User:</b>",
                username,
            )
```

File: openpype/tools/workfiles/window.py (hunk 3 above)
Comment (BigRoy) [reply openpype/tools/workfiles/window.py:108]: `pwd` also works on Mac right?

Comment (fabiaserra) [reply openpype/tools/workfiles/window.py:108]: yeah it definitely works in Mac as well, just tested it on a Mac of ours:
```
vfx-studiomac-82:/ farrizabalaga$ python
Python 2.7.16 (v2.7.16:413a49145e, Mar  2 2019, 14:32:10) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pwd
>>> import os
>>> filestat = os.stat("/Scripts/mount_transfer_storage.sh")
>>> pwd.getpwuid(filestat.st_uid).pw_name
'root'
```

File: openpype/tools/workfiles/window.py (hunk 4 above)
Comment (fabiaserra) [reply openpype/tools/workfiles/window.py:156]: sure, commiting your suggestion

File: openpype/tools/workfiles/window.py (hunk 2 above)
Comment (fabiaserra) [reply openpype/tools/workfiles/window.py:105]: yep

File: openpype/tools/workfiles/window.py (hunk 3 above)
Comment (iLLiCiTiT) [reply openpype/tools/workfiles/window.py:108]: Ok, changed the condition to match the usecase.

File: openpype/tools/workfiles/window.py [hunk 5]
```diff
@@ -94,6 +95,19 @@ def _on_save_click(self):
         self._on_note_change()
         self.save_clicked.emit()
 
+    def get_user_name(self, file):
+        """Get user name from file path"""
+        # Only run on Unix because pwd module is not available on Windows.
+        # NOTE: we tried adding "win32security" module but it was not working
+        # on all hosts so we decided to just support Linux until migration
+        # to Ayon
+        if platform.system().lower() == "window":
```
Comment (iLLiCiTiT) [inline openpype/tools/workfiles/window.py:104]: And I've made a typo 😞 
```suggestion
        if platform.system().lower() == "windows":
```

Comment (fabiaserra) [reply openpype/tools/workfiles/window.py:104]: hehe, good catch!

**Review by iLLiCiTiT — APPROVED** [review]
LGTM. Don't have linux or mac to test.

**Comment by fabiaserra** [comment]
> LGTM. Don't have linux or mac to test.

I can confirm it still works with the current changes on my Linux box.

Re: mac I don't have OP set up there but I could confirm that `pwd` exists and the function works

**Comment by mkolar** [comment]
Thank you @fabiaserra merging.

**Pull request merged**